### PR TITLE
[TrueHD] Fix CEngineStats delay/sync in TrueHD IEC + PAPlayer fixes

### LINKED ISSUE #20391: Atmos Music: Progressbar and Duration stutter issue
### Describe the bug
Here is a clear and concise description of what the problem is:

When playing Atmos music (audio only) in Kodi 18 and 19 (all versions) the song progressbar and duration timer both stutter badly moving forward and backward very fast during playback. This occurs on Estuary and Aeon Nox Silvo skins. When using HDMI pass-through to an AVR for Atmos decoding.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Both progressbar and duration time should flow smoothly like any other codec when played.

## Actual Behavior

Progressbar and duration jumps. Duration is showing changes in the numeric value multiple times per second. Duration should increment once per second,

### To Reproduce

1. Configure Kodi to play via HDMI Pass-through and enable that your AVR can decode TrueHD.
2. Play an (audio) song file with Atmos codec. Switch to a view showing either progressbar or duration timer.

Download sample here:
https://drive.google.com/drive/folders/1QQg3yAfZyyCRI2oPxpjqmRG_1dMQ23tD?usp=sharing

NOTE 1: Both sets of MKA and M4A files contain Atmos codecs.
NOTE2 : The Abbey Road samples will only play back with latest Kodi builds (a recent Kodi high bitrate fix):
https://github.com/xbmc/xbmc/pull/20339

### Debuglog

The debuglog can be found here:

### Screenshots 
Here are some links or screenshots to help explain the problem:

No screen shot will show this as it needs an animation

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [x ] Android
 - [ ] iOS
 - [ ] tvOS
 - [x ] Linux (LibreEelc)
 - [ ] OSX
 - [x ] Windows
 - [ ] Windows UWP

I have personally had this issue in all Kodi versions from 18 to 19.1 using LibreElec (Intel NUCs).
I have seen reports of users with Android / NVidia Shield Kodi 19.2 . I had this issue with Kodi 18 on Windows (NUC)
I do not have this problem on Windows 19 on my PC without HDMI pass-through (i.e. Kodi decoding Atmos)

 - Operating system version/name:
 - Kodi version:
 - 19.0, 19.1, 19.2
 - 18.4, 18.5 etc

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

### PR DESCRIPTION
**Pull request by thexai**
## Description
Fix CEngineStats delay/sync in TrueHD IEC + PAPlayer fixes

- Fixes Debug Info OSD is imprecise (oscillate) witch TrueHD IEC.
- Fixes PAPlayer progress indicator is imprecise (oscillate) witch TrueHD IEC.
- Fixes PAPlayer progress indicator is wrong duration in Android using TrueHD RAW.
- Improves `CAEStreamInfo::GetDuration()` parameter name.
- Closes https://github.com/xbmc/xbmc/issues/20391

## Motivation and context
While this not fixes any TrueHD dropout, some things still wrong related to TrueHD. This tries to fix a few detected issues...

## How has this been tested?
Runtime tested Windows 11 + NUCi3BEK + Denon X1600H Video player and PAPlayer (.mka THD file)
Runtime tested Nvidia Shield Pro IEC + RAW

## What is the effect on users?
See description

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp [hunk 1]
```diff
@@ -73,7 +73,7 @@ void CEngineStats::GetDelay(AEDelayStatus& status)
   if (m_pcmOutput)
     status.delay += (double)m_bufferedSamples / m_sinkSampleRate;
   else
-    status.delay += (double)m_bufferedSamples * m_sinkFormat.m_streamInfo.GetDuration() / 1000;
+    status.delay += (double)m_bufferedSamples * m_sinkFormat.m_streamInfo.GetDuration(m_sinkNeedIecPack) / 1000;
```
Comment (enen92) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp:76]: While you are at it maybe use a c++ style cast for the changed line?

File: xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp [hunk 2]
```diff
@@ -144,7 +145,8 @@ void CEngineStats::GetDelay(AEDelayStatus& status, CActiveAEStream *stream)
   if (m_pcmOutput)
     status.delay += (double)m_bufferedSamples / m_sinkSampleRate;
   else
-    status.delay += (double)m_bufferedSamples * m_sinkFormat.m_streamInfo.GetDuration() / 1000;
+    status.delay +=
```
Comment (enen92) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp:148]: c style cast

File: xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp [hunk 3]
```diff
@@ -167,7 +169,8 @@ void CEngineStats::GetSyncInfo(CAESyncInfo& info, CActiveAEStream *stream)
...
```
Comment (enen92) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp:172]: Same

File: xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAESink.h [hunk 4]
```diff
@@ -100,6 +100,7 @@ class CActiveAESink : private CThread
   bool HasPassthroughDevice();
   bool SupportsFormat(const std::string &device, AEAudioFormat &format);
   bool DeviceExist(std::string driver, const std::string& device);
+  bool NeedIecPack() { return m_needIecPack; }
```
Comment (lrusak) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAESink.h:103]:
```suggestion
  bool NeedIecPack() const { return m_needIecPack; }
```

**Review by fritsch — APPROVED** [review]
Thanks much. Let's see how many we continue to find.

**Comment by thexai** [comment]
Merge now to be included in Jan-21 build :)

**Pull request merged**